src/internal/storage/Storage.h

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <concepts>
+#include <functional>
+#include <string>
+
+#include <external/expected/Expected.h>
+
+namespace internal::storage {
+
+class Position {
+public:
+	Position() = default;
+	Position(const Position &) = default;
+
+	/*
+	 * Intentionally implicit
+	 */
+	Position(long pos) : m_pos{pos}, m_isset{true} {}
```

Comment:
Single argument constructors should be marked explicit

Reply:
Yes, but this is supposed to the just that - implicitly convert `long`s to `Position`.